external_parser/example_joiner.h

```diff
@@ -4,30 +4,30 @@
 
 #include "example.h"
 #include "lru_dedup_cache.h"
-#include "reward.h"
+#include "i_joiner.h"
 #include "v_array.h"
 
 #include <fstream>
 #include <list>
 #include <queue>
 #include <unordered_map>
 
-class example_joiner {
+class example_joiner : public i_joiner {
 public:
   example_joiner(vw *vw); // TODO rule of 5
   example_joiner(vw *vw, bool binary_to_json, std::string outfile_name);
 
-  ~example_joiner();
+  virtual ~example_joiner();
```

Comment:
nit: I also think you don't need virtual on the destructor either, I think it is explicitly `virtual` since the base class has it's destructor virtual

Reply:
yeah technically i do not, I am just having habit of always being explicit about virtual methods.

Reply:
ok no problem